pkg/pipeline/conntrack_integ_test.go

```diff
@@ -0,0 +1,154 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package pipeline
+
+import (
+	"bufio"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/decode"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/ingest"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/write"
+	"github.com/netobserv/flowlogs-pipeline/pkg/test"
+	"github.com/stretchr/testify/require"
+)
+
+const testConfigConntrack = `---
+pipeline:
+  - name: ingest_fake
+  - follows: ingest_fake
+    name: conntrack
+  - follows: conntrack
+    name: write_fake
+parameters:
+  - ingest:
+      type: fake
+    name: ingest_fake
+  - name: conntrack
+    track:
+      type: conntrack
+      conntrack:
+        endConnectionTimeout: 1s
+        outputRecordTypes:
+          - newConnection
+          - flowLog
+          - endConnection
+        keyDefinition:
+          fieldGroups:
+            - name: src
+              fields:
+                - SrcAddr
+                - SrcPort
+            - name: dst
+              fields:
+                - DstAddr
+                - DstPort
+            - name: protocol
+              fields:
+                - Proto
+          hash:
+            fieldGroupRefs: [protocol]
+            fieldGroupARef: src 
+            fieldGroupBRef: dst 
+        outputFields:
+          - name: Bytes
+            operation: sum
+            SplitAB: true
+          - name: Packets
+            operation: sum
+            SplitAB: true
+          - name: numFlowLogs
+            operation: count
+          - name: TimeFlowStart
+            operation: min
+            input: TimeReceived
+          - name: TimeFlowEnd
+            operation: max
+            input: TimeReceived
+  - name: write_fake
+    write:
+      type: fake
+`
+
+func TestConnTrack(t *testing.T) {
+	// This test runs a 3 stage pipeline (Ingester -> ConnTrack -> Write), ingests a file and tests that an end
+	// connection record with specific values was written.
+	var mainPipeline *Pipeline
+	var err error
+	v := test.InitConfig(t, testConfigConntrack)
+	require.NotNil(t, v)
+
+	mainPipeline, err = NewPipeline()
+	require.NoError(t, err)
+
+	go mainPipeline.Run()
+
+	in := mainPipeline.pipelineStages[0].Ingester.(*ingest.IngestFake).In
+	writer := mainPipeline.pipelineStages[2].Writer.(*write.WriteFake)
+
+	ingestFile(t, in, "../../hack/examples/ocp-ipfix-flowlogs.json")
+	writer.Wait()
+	writer.ResetWait()
+
+	// Wait a moment to make the connections expired
+	time.Sleep(2 * time.Second)
```

Comment:
I'm not very pleased with this 2 second sleep. It obviously slows the test and on if the running platforms is loaded it might fail.
I can remediate this by injecting a fake clock and simulate the sleep. But, on the other hand, I think of this test as an integration test and I'm not sure if we want to fake the clock. WDYT?